modules/couchbase/couchbase.go

```diff
@@ -0,0 +1,593 @@
+		response, err := c.doHttpRequest(ctx, QUERY_PORT, "/query/service", http.MethodPost, body, true)
+		if err != nil {
+			return err
+		}
+
+		online := gjson.Get(string(response), "results.0.online").Bool()
+		if !online {
+			return errors.New("primary index state is not online")
+		}
+
+		return nil
+	}, backoff.WithContext(backoff.NewExponentialBackOff(), ctx))
+
+	return err
+}
+
+func (c *CouchbaseContainer) createPrimaryIndex(ctx context.Context, bucket bucket) error {
+	body := map[string]string{
+		"statement": "CREATE PRIMARY INDEX on `" + bucket.name + "`",
+	}
+
+	_, err := c.doHttpRequest(ctx, QUERY_PORT, "/query/service", http.MethodPost, body, true)
+
+	return err
+}
+
+func (c *CouchbaseContainer) isQueryKeyspacePresent(ctx context.Context, bucket bucket) error {
+	body := map[string]string{
+		"statement": "SELECT COUNT(*) > 0 as present FROM system:keyspaces WHERE name = \"" + bucket.name + "\"",
+	}
+
+	err := backoff.Retry(func() error {
+		response, err := c.doHttpRequest(ctx, QUERY_PORT, "/query/service", http.MethodPost, body, true)
+		if err != nil {
+			return err
+		}
+		present := gjson.Get(string(response), "results.0.present").Bool()
+		if !present {
+			return errors.New("query namespace is not present")
+		}
+
+		return nil
+	}, backoff.WithContext(backoff.NewExponentialBackOff(), ctx))
+
+	return err
+}
+
+func (c *CouchbaseContainer) waitForAllServicesEnabled(ctx context.Context, bucket bucket) error {
+	err := wait.ForHTTP("/pools/default/b/"+bucket.name).
+		WithPort(MGMT_PORT).
+		WithBasicAuth(c.config.username, c.config.password).
+		WithStatusCodeMatcher(func(status int) bool {
+			return status == http.StatusOK
+		}).
+		WithResponseMatcher(func(body io.Reader) bool {
+			response, err := io.ReadAll(body)
+			if err != nil {
+				return false
+			}
+			return c.checkAllServicesEnabled(response)
+		}).
+		WaitUntilReady(ctx, c)
+
+	return err
+}
+
+func (c *CouchbaseContainer) createBucket(ctx context.Context, bucket bucket) error {
+	flushEnabled := "0"
+	if bucket.flushEnabled {
+		flushEnabled = "1"
+	}
+	body := map[string]string{
+		"name":          bucket.name,
+		"ramQuotaMB":    strconv.Itoa(bucket.quota),
+		"flushEnabled":  flushEnabled,
+		"replicaNumber": strconv.Itoa(bucket.numReplicas),
+	}
+
+	_, err := c.doHttpRequest(ctx, MGMT_PORT, "/pools/default/buckets", http.MethodPost, body, true)
+
+	return err
+}
+
+func (c *CouchbaseContainer) doHttpRequest(ctx context.Context, port, path, method string, body map[string]string, auth bool) ([]byte, error) {
+	form := url.Values{}
+	for k, v := range body {
+		form.Set(k, v)
+	}
+
+	url, err := c.getUrl(ctx, port, path)
+	if err != nil {
+		return nil, err
+	}
+
+	request, err := http.NewRequestWithContext(ctx, method, url, strings.NewReader(form.Encode()))
+	if err != nil {
+		return nil, err
+	}
+
+	request.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+
+	if auth {
+		request.SetBasicAuth(c.config.username, c.config.password)
+	}
```

Comment:
Sure, my point is, instead of passing `auth bool`:

```go
if config.username != "" && config.password != "" {
    request.SetBasicAuth(config.username, config.password)
}
```

passing the config instead of the boolean.

But in any case this is a nit, I'd prefer avoiding calls passing `true|false`, but it'd be fine if we all agree in keeping this method signature.

Reply:
Ohh I see now, there are methods using basic auth and others not using it. Got it, then I think it's fine keeping it as is

Thanks for the feedback!